macros/get_lookback_date_limits.sql

```diff
@@ -1,17 +1,41 @@
-{% macro get_lookback_date_limits(limit_type) %}
-  {{ return(adapter.dispatch('get_lookback_date_limits', 'snowplow_fractribution')(limit_type)) }}
+{% macro get_lookback_date_limits(limit_type, model) %}
+  {{ return(adapter.dispatch('get_lookback_date_limits', 'snowplow_fractribution')(limit_type, model)) }}
 {% endmacro %}
 
-{% macro default__get_lookback_date_limits(limit_type) %}
-
-  -- check if web data is up-to-date
+{% macro default__get_lookback_date_limits(limit_type, model) %}
 
+  -- check if web data is up-to-date in page_views_source (should cover conversion source check in case the web model is used to model conversions)
+  {% set combined_time = var("snowplow__conversion_window_days") + var('snowplow__path_lookback_days') %}
   {% set query %}
-    select max(start_tstamp) < '{{ var('snowplow__conversion_window_end_date') }}' as is_over_limit,
-           cast(min(start_tstamp) as date) > '{{ var("snowplow__conversion_window_start_date") }}' as is_below_limit,
-           cast(max(start_tstamp) as {{ type_string() }}) as last_processed_page_view,
-           cast(min(start_tstamp) as {{ type_string() }}) as first_processed_page_view
-    from {{ var('snowplow__page_views_source') }}
+
+      -- when the user opts for the auto-populated conversion window
+      {% if var("snowplow__conversion_window_start_date") == '' and var("snowplow__conversion_window_end_date") == '' %}
+           with base as (
+              select max(start_tstamp) as last_pageview,
+                    min(start_tstamp) as first_pageview,
+              from {{ var('snowplow__page_views_source') }}
+           )
+           select
+             false as is_over_limit, -- the last pageview will be taken from the page_views_source
+             cast(first_pageview as date) > cast({{ dbt.dateadd('day', -combined_time, last_pageview) }} as date) as is_below_limit,
```

Comment:
Just to confirm across all warehouses that cast to date will remove the time AND truncate down?

Reply:
Tested it for each:

```
--redshift / postgres / snowflake / bigquery / databricks
select cast(current_timestamp as date)
```

All resulted in '2023-08-16'